sympy/sets/handlers/intersection.py

```diff
@@ -306,7 +306,7 @@ def intersection_sets(self, other): # noqa:F811
             # univarite imaginary part in same variable
             x, xis = zip(*[solve_linear(i, 0) for i in Mul.make_args(im) if n in i.free_symbols])
             if x and all(i == n for i in x):
-                base_set -= FiniteSet(xis)
+                base_set = FiniteSet(*xis)
             else:
                 base_set -= ConditionSet(n, Eq(im, 0), S.Integers)
```

Comment:
```suggestion
                base_set = ConditionSet(n, Eq(im, 0), base_set)
```

Reply:
I suspect the same change should be made below, too, at `base_set -= ConditionSet(n, Eq(i, 0), S.Integers)`, changing `S.Integers` to `base_set`. And in these cases `-=`, not `&=` is correct, right, because we are removing values of n that are invalid.